aiida/repository/control.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Module for overall repository control commands."""
+# Note: these functions are not methods of `AbstractRepositoryBackend` because they need access to the orm.
+# This is because they have to go through all the nodes to gather the list of keys that AiiDA is keeping
+# track of (since they are descentralized in each node entry).
+# See the get_unreferenced_keyset function
+from aiida.manage.manager import get_manager
+from aiida.orm.implementation import Backend
+
+
+def repository_maintain(full: bool = False, backend: Backend = None, **kwargs) -> dict:
+    """Performs maintenance tasks on the repository."""
+    maintainance_report = {'database': {}, 'repository': {}, 'user_info': ''}
+
+    if backend is None:
+        backend = get_manager().get_backend()
+    repository_backend = backend.get_repository()
+
+    unreferenced_objects = get_unreferenced_keyset(aiida_backend=backend)
+
+    maintainance_report['repository']['unreferenced'] = len(unreferenced_objects)  # type: ignore
+    maintainance_report['repository']['info'] = repository_backend.get_info()  # type: ignore
+
+    repository_backend.delete_objects(list(unreferenced_objects))
+
+    # Perform the maintainance operations in the repository
+    maintainance_report['repository']['maintain'] = repository_backend.maintain(full=full, **kwargs)  # type: ignore
```

Comment:
I had to disable mypy here because I am getting errors I don't understand:

```
aiida/repository/control.py:29: error: Unsupported target for indexed assignment ("Collection[str]") [index] 
aiida/repository/control.py:30: error: Unsupported target for indexed assignment ("Collection[str]") [index] 
aiida/repository/control.py:35: error: Unsupported target for indexed assignment ("Collection[str]") [index] 
```